src/Generator.cpp

```diff
@@ -1004,6 +1004,9 @@ int generate_filter_main(int argc, char **argv, std::ostream &cerr) {
                     // Must re-create each time since each instance will have a different Target.
                     auto gen = GeneratorRegistry::create(generator_name, GeneratorContext(target));
                     gen->set_generator_param_values(sub_generator_args);
+                    // Reset the counters so the function/variable names look
+                    // more consistent for different targets.
+                    reset_unique_name_counters();
```

Comment:
This is not quite safe, because code in another thread could create pipelines with duplicated names in it if this runs while it is being defined. I'm sure there's a simple solution to getting consistent autogenerated names per generator invocation, but it's eluding me. Maybe some sort of thread-local naming context? Zalman or Steven might have an idea.

Reply:
In all the current AOT build systems I'm aware of, there won't be multiple threads in the same process targeting this.... but yeah, it's a bug waiting to happen. Making the unique-name-counter stuff use thread locals would probably be a safe and simple fix (at a nonzero but probably small runtime cost.)